guides/keras_cv/object_detection_keras_cv.py

```diff
@@ -0,0 +1,868 @@
+weights to weights produced by training a classification model on the imagenet
+dataset.
+
+KerasCV makes it easy to construct a `RetinaNet` with any of the KerasCV
+backbones.  Simply use one of the presets for the architecture you'd like!
+
+For example:
+"""
+
+model = keras_cv.models.RetinaNet.from_preset(
+    "resnet50_imagenet",
+    # number of classes to be used in box classification
+    num_classes=len(class_mapping) + 1,
+    # For more info on supported bounding box formats, visit
+    # https://keras.io/api/keras_cv/bounding_box/
+    bounding_box_format="xywh",
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts
+tuples of dense image Tensors and bounding box dictionaries to `fit()` and
+`train_on_batch()`
+
+This matches what we have constructed in our input pipeline above.
+
+**Note: be sure to freeze BatchNormalization layers when training a RetinaNet.***
+
+And important but easy to miss step in training your own object detection model
+is freezing the `BatchNormalization` layers in your backbone.
+We felt it would be confusing to do this automatically,
+so instead we recommend you do it in your training loops.
+"""
+
+for layer in model.backbone.layers:
+    if isinstance(layer, keras.layers.BatchNormalization):
+        layer.trainable = False
+
+"""
+
+### Optimizer
+
+In this guide, we use a standard SGD optimizer and rely on the [`keras.callbacks.ReduceLROnPlateau`](https://keras.io/api/callbacks/reduce_lr_on_plateau/) callback to reduce the learning rate.
+
+You will always want to include a `global_clipnorm` when training object detection models.  This is to remedy exploding gradient problems that frequently occur when training object detection models.
+"""
+
+base_lr = 0.005
+# including a global_clipnorm is extremely important in object detection tasks
+optimizer = tf.keras.optimizers.SGD(
+    learning_rate=base_lr, momentum=0.9, global_clipnorm=10.0
+)
+
+"""
+To achieve the best results on your dataset, you'll likely want to hand craft a
+`PiecewiseConstantDecay` learning rate schedule.
+While `PiecewiseConstantDecay` schedules tend to perform better, they don't
+translate between problems.
+"""
+
+"""
+### Loss Functions
+
+You may not be familiar with the `"focal"` or `"smoothl1"` losses.  While not common in other models, these losses are more or less staples in the object detection world.
+
+In short, ["Focal Loss"](https://arxiv.org/abs/1708.02002) places extra emphasis on difficult training examples.  This is useful when training the classification loss, as the majority of the losses are assigned to the background class.
+
+"SmoothL1 Loss" is used to [prevent exploding gradients](https://arxiv.org/abs/1504.08083) that often occur when attempting to perform the box regression task.
+
+In KerasCV you can use these losses simply by passing the strings `"focal"` and `"smoothl1"` to `compile()`:
+"""
+
+pretrained_model.compile(
```

Comment:
`model.compile(...)`?

Reply:
Now that I moved this model doesn't exist anymore at this cell 👍